### DISCUSSION TITLE
Gratis Bio Menace and more DOS games on GOG

### DISCUSSION THREAD

**Discussion by jschwartzenberg** [dosemu2/dosemu2#1575] [General]
For a few days there GOG is offering multiple games for free. At least Bio Menace includes DOS binaries. It seems to hang at start-up though:
![Screenshot_20220107_195312](https://user-images.githubusercontent.com/877112/148592656-37cb2b40-6995-4f78-ba2c-54b386906efa.png)

**Comment by stsp** [comment]
Please try on cpu-emu.

**Reply by jschwartzenberg** [reply]
Just tried, same result.

**Reply by stsp** [reply]
[bio.tar.gz](https://github.com/dosemu2/dosemu2/files/7833101/bio.tar.gz)
Please try this.
You need to update to latest
devel (or sdl will crash) and
need to run biopatch.exe before
running game, every time.

**Reply by stsp** [reply]
And you need to unpack the
archive attached to prev comment
(in case the archive wasn't noticed)

**Reply by jschwartzenberg** [reply]
"Dosbox Bio Menace 1-2-3 patch active." Is there anything specific to dosbox about this version of the game? I just realized there was also this issue in the past, #905. Is this related?

Indeed the game runs now! As it's based around the Commander Keen engine, #501 is there.

**Comment by jschwartzenberg** [comment]
Other DOSBox games that are free now and thus have DOS binaries:
https://www.gog.com/game/akalabeth_world_of_doom
https://www.gog.com/game/jill_of_the_jungle_the_complete_trilogy
https://www.gog.com/game/worlds_of_ultima_the_savage_empire
https://www.gog.com/game/ultima_worlds_of_adventure_2_martian_dreams
https://www.gog.com/game/ultima_4

**Reply by jschwartzenberg** [reply]
I forgot the original URL, here's the full list: https://www.gog.com/partner/free_games

**Comment by stsp** [comment]
No, #905 is unrelated to dosbox patch.
The patch for 905 is embedded in the
binaries that I uploaded.
I dont know what dosbox patch does but
the game is unplayable otherwise.
So there are 2 patches here and a remap.c
patch in devel to get that working.
There is also a horizontal bar glitch.
If you claim the ladder you'll eventually
see the undrawn horizontal line.
You can open a ticket for that if you want.

**Reply by jschwartzenberg** [reply]
> I dont know what dosbox patch does but
> the game is unplayable otherwise.

But where did you get `biopatch.exe` from that tarball from?

**Reply by stsp** [reply]
Just typed "Gratis Bio Menace" in
ddg and used the first link to D/L the
game. There was such patch.

**Reply by jschwartzenberg** [reply]
It's interesting that this patch is not included in the GOG release and runs in its DOSBox version without it. https://www.gog.com/game/bio_menace

**Reply by stsp** [reply]
What is the size of an exe files?
Maybe they applied some other patch.

**Reply by stsp** [reply]
Or maybe they patched dosbox.
Who knows. :)

**Reply by jschwartzenberg** [reply]
You can still get the game from GOG ;)
```
[julius@elleboog data]$ pwd
/home/julius/games/GOG Games/Bio Menace/data
[julius@elleboog data]$ ls -l
totaal 3076
-rw-r--r-- 1 julius julius    580  7 jan 19:38 AUDIOHED.BM1
-rw-r--r-- 1 julius julius    564  7 jan 19:38 AUDIOHED.BM2
-rw-r--r-- 1 julius julius    552  7 jan 19:38 AUDIOHED.BM3
-rw-r--r-- 1 julius julius 205754  7 jan 19:38 AUDIOT.BM1
-rw-r--r-- 1 julius julius 244224  7 jan 19:38 AUDIOT.BM2
-rw-r--r-- 1 julius julius 201662  7 jan 19:38 AUDIOT.BM3
-rw-r--r-- 1 julius julius   2207  7 jan 19:38 BIOPATCH.ZIP
-rw-r--r-- 1 julius julius  90046  7 jan 19:38 BMENACE1.EXE
-rw-r--r-- 1 julius julius  89921  7 jan 19:38 BMENACE2.EXE
-rw-r--r-- 1 julius julius  89754  7 jan 19:38 BMENACE3.EXE
-rw-r--r-- 1 julius julius  11627  7 jan 19:38 BMENACE.conf
-rw-r--r-- 1 julius julius  15076  7 jan 19:38 BM-HELP.TXT
-rw-r--r-- 1 julius julius  11653  7 jan 19:38 BM-HINT.TXT
-rw-r--r-- 1 julius julius    543  8 jan 01:03 CONFIG.BM1
-rw-r--r-- 1 julius julius    543  7 jan 19:38 CONFIG.BM2
-rw-r--r-- 1 julius julius   1024  7 jan 19:38 EGADICT.BM1
-rw-r--r-- 1 julius julius   1024  7 jan 19:38 EGADICT.BM2
-rw-r--r-- 1 julius julius   1024  7 jan 19:38 EGADICT.BM3
-rw-r--r-- 1 julius julius 501816  7 jan 19:38 EGAGRAPH.BM1
-rw-r--r-- 1 julius julius 479241  7 jan 19:38 EGAGRAPH.BM2
-rw-r--r-- 1 julius julius 412364  7 jan 19:38 EGAGRAPH.BM3
-rw-r--r-- 1 julius julius  13866  7 jan 19:38 EGAHEAD.BM1
-rw-r--r-- 1 julius julius  13947  7 jan 19:38 EGAHEAD.BM2
-rw-r--r-- 1 julius julius  12693  7 jan 19:38 EGAHEAD.BM3
-rw-r--r-- 1 julius julius    442  7 jan 19:38 FILE_ID.DIZ
-rw-r--r-- 1 julius julius   1658  7 jan 19:38 GAMEMAPS.BM1
-rw-r--r-- 1 julius julius   1318  7 jan 19:38 GAMEMAPS.BM2
-rw-r--r-- 1 julius julius     11  7 jan 19:38 GAMEMAPS.BM3
-rw-r--r-- 1 julius julius     46  7 jan 19:38 GFXINFOE.BM1
-rw-r--r-- 1 julius julius     46  7 jan 19:38 GFXINFOE.BM2
-rw-r--r-- 1 julius julius     46  7 jan 19:38 GFXINFOE.BM3
-rw-r--r-- 1 julius julius  22200  7 jan 19:38 MAPHEAD.BM1
-rw-r--r-- 1 julius julius  21120  7 jan 19:38 MAPHEAD.BM2
-rw-r--r-- 1 julius julius  21336  7 jan 19:38 MAPHEAD.BM3
-rw-r--r-- 1 julius julius 165402  7 jan 19:38 MAPTEMP.BM1
-rw-r--r-- 1 julius julius 196526  7 jan 19:38 MAPTEMP.BM2
-rw-r--r-- 1 julius julius 197166  7 jan 19:38 MAPTEMP.BM3
-rw-r--r-- 1 julius julius   4234  7 jan 19:38 MAPTHEAD.BM1
-rw-r--r-- 1 julius julius   4325  7 jan 19:38 MAPTHEAD.BM2
-rw-r--r-- 1 julius julius   4412  7 jan 19:38 MAPTHEAD.BM3

**Comment by stsp** [comment]
So you already have biopatch.zip

**Reply by jschwartzenberg** [reply]
Ah wow, I hadn't noticed the zip file :)

It doesn't seem to get used though. Apparently the patch comes from the DOSBox team according to the included readme.

**Comment by stsp** [comment]
How come it doesn't get used?
You need to run it or game doesn't work.

**Reply by jschwartzenberg** [reply]
It also seems to work without in newer versions of dosbox. I can just run the executable directly. I suspect the fix was needed in older dosbox versions.

**Reply by stsp** [reply]
It will work but it eventually crashes.

**Reply by stsp** [reply]
I mean, it eventually crashes
on dosemu2. Maybe it doesn't
on newer dosbox. 5 minutes of
a gameplay should be enough
to check that.

**Reply by jschwartzenberg** [reply]
I guess it's fixed in newer dosbox versions. The start-up of the game doesn't include anything to run this biopatch and I could play around 4-5 levels without running into a crash.

**Reply by stsp** [reply]
Maybe you can find some info
about that patch on their forum. :)
Once we know what it does, we
can decide if dosemu2 can be
"fixed" the same way.
